integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/AppendDataDatasetBuilder.java

```diff
@@ -34,12 +34,7 @@ public boolean isDefinedAtLogicalPlan(LogicalPlan logicalPlan) {
   }
 
   @Override
-  public List<OpenLineage.OutputDataset> apply(AppendData x) {
-    return Collections.emptyList();
-  }
-
-  @Override
-  public List<OpenLineage.OutputDataset> apply(SparkListenerEvent event, AppendData appendData) {
+  protected List<OpenLineage.OutputDataset> apply(SparkListenerEvent event, AppendData appendData) {
```

Comment:
_Minor_: Since `apply()` was scoped as `public` in `AppendDataDatasetBuilder`, any concerns in limiting the scope to `protected`? We'll want to document the method scope change of the class in our changelog as this would be a breaking change.